eliot/_action.py

```diff
@@ -415,6 +424,286 @@ def __exit__(self, type, exception, traceback):
         self.finish(exception)
 
 
+class WrongTask(Exception):
+    """
+    Tried to add a message to an action, but the message was from another
+    task.
+    """
+
+    def __init__(self, action, message):
+        Exception.__init__(
+            self, 'Tried to add {} to {}. Expected task_uuid = {}, got {}'.format(
+                message, action, action.task_uuid, message.task_uuid))
+
+
+class WrongTaskLevel(Exception):
+    """
+    Tried to add a message to an action, but the task level of the message
+    indicated that it was not a direct child.
+    """
+
+    def __init__(self, action, message):
+        Exception.__init__(
+            self, 'Tried to add {} to {}, but {} is not a sibling of {}'.format(
+                message, action, message.task_level, action.task_level))
+
+
+class WrongActionType(Exception):
+    """
+    Tried to end a message with a different action_type to the beginning.
```

Comment:
It's [a regional thing](http://www.oxforddictionaries.com/words/different-from-than-or-to). I find "different than" awkward, but happy to standardize on US English.